packages/js/product-editor/src/hooks/use-product-helper.ts

```diff
@@ -25,11 +21,14 @@ import { CurrencyContext } from '@woocommerce/currency';
 /**
  * Internal dependencies
  */
+import { AUTO_DRAFT_NAME, getDerivedProductType } from '../index';
 import {
 	NUMBERS_AND_DECIMAL_SEPARATOR,
 	ONLY_ONE_DECIMAL_SEPARATOR,
-} from './constants';
-import { ProductVariationsOrder } from './hooks/use-variations-order';
+} from '../constants';
+import { ProductVariationsOrder } from './use-variations-order';
+
+// TODO: Having to add TS ignore comments here, should be able to address with TS config but running into issues.
```

Comment:
I'd like to remove these `ts-ignore` lines, as they weren't needed when the code resided within WCA. I tried fiddling around with the `lib` and `target` props in the relevant `tsconfig.json` files, to no avail. In some cases the linting would be fixed but not the build process, and in most cases it just adds more errors involving unknown globals (such as `window`) if I add `es2021` to the `lib` property for example.

Any ideas are welcome!

Reply:
Hmm yeah that's because we still support `es6` for our build files, I am curious if it would be worth updating this to `es2020` or `es2021` as you mentioned. This might be worth an RFC and I wonder if it would save us some bundle space as well if we update this.
Curious what the standard for Gutenberg around this is.

Anyhow, I am ok with using these comments for now, but in theory we may want to solve this by replacing `flatMap` and `replaceAll` with `replace` using the global match and maybe the `flatMap` from lodash (or seeing if we can just use `map` or `reduce` some how.
Given this is a new brand new package its probably ok as is.

Reply:
>  we may want to solve this by replacing flatMap and replaceAll with replace using the global match and maybe the flatMap from lodash (or seeing if we can just use map or reduce some how.

Right, considered doing so although it seems odd that we have a difference configuration in the `/client` code where this originated without issue.

Reply:
Yeah that's why I would propose an RFC to address this.